addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1086,7 +1095,8 @@ const Wysiwyg = Widget.extend({
      * Stop the field changes mutation observers.
      */
     _pauseOdooFieldObservers: function () {
-        for (let observerData of this.odooFieldObservers) {
+        const fieldObserversData = this.__externalOdooFieldObservers || this.odooFieldObservers;
```

Comment:
I don't like this line. First, I think `this.__externalOdooFieldObservers` is quite confusing. I don't understand why those observers are more "external" than the others just because they aren't inner translations. Maybe it makes sense and I'm the only one confused by this, I don't know, but when I see this line, my first thought is "but wait, what about the other observers, why don't we pause them?". It just so happens that it's exactly what your commit aims to do, but having a function called `_pauseOdooFieldObservers` that completely ignores `this.odooFieldObservers` if `this.__externalOdooFieldObservers` is defined is very confusing to me.

Moreover, are you sure that it works when there are no observer in `this.__externalOdooFieldObservers`? Because an empty array is still truthy so unless I'm missing something, `this.odooFieldObservers` will never be paused, even when no observer actually matched the filter for `this.__externalOdooFieldObservers`.

What I would do to solve both issues is to explicitely filter here. So it would be:
```js
_pauseOdooFieldObservers: function () {
    observersToPause = this.odooFieldObservers.filter(
        // Exclude inner translation fields observers. They
        // still handle translation synchronization inside the
        // targeted field.
        observerData => !observerData.field.dataset.oeTranslationInitialSha ||
            !field.contains(observerData.field)
    );
    ...
}
```
I don't know if the comment still applies or not in this case as I'm not sure what you meant by it. However, I have a feeling the comment was there _specifically_ because you already tried what I just suggested and it didn't work. Let me know if that's the case.

Reply:
@dmo-odoo  Thanks for the check :+1: To answer your questions:

> I don't understand why those observers are more "external" than the others just because they aren't inner translations...

A- It was actually used to reference (temporarily, since it was added like that only for compatibility in stable and will be removed in master) the observers linked to targets “external” to the current field itself… I'm changing  it to `__odooFieldObserversToPause` which has a more direct meaning :+1:.

> but having a function called _pauseOdooFieldObservers that completely ignores this.odooFieldObservers if this.__externalOdooFieldObservers is defined is very confusing to me...

B- This is also a [compatibility adaptation](https://github.com/odoo/odoo/pull/176198#discussion_r1719957504): we don't want to break the current behavior if a custom code in stable is using `_pauseOdooFieldObservers()` without defining `this.__externalOdooFieldObservers`. So we use `this.odooFieldObservers` as fallback. I added a comment to clarify that :+1:.

> Moreover, are you sure that it works when there are no observer in this.__externalOdooFieldObservers?...


C- We always target `this.__externalOdooFieldObservers`, `this.odooFieldObservers` is just a fallback in case it was not defined (see [B]). 

> I don't know if the comment still applies or not in this case as I'm not sure what you meant by it. However, I have a feeling the comment was there specifically because you already tried what I just suggested and it didn't work. Let me know if that's the case.

D- In the code you suggested, you'll need the current updated `field` to get the observers of targets outside of it… Actually, that's the whole point of writing the code that way: to avoid changing the signature of `_pauseOdooFieldObservers()` in stable (which was mentioned in the TODO comment: **in master, we will simply call `this._pauseOdooFieldObservers(field)` with the code you suggested** :+1:).

What do you think ?

Reply:
A, D: 👍
B, C: I see what you mean, but I don't really understand the cases where it won't be defined since it does not seem to be an actual option that you can pass?

Reply:
@dmo-odoo If someone is using `_pauseOdooFieldObservers()` in some kind of custom code [A], he won't need to do anything since it's supposed to handle all observers (`this.odooFieldObservers`). But after this diff, the `_pauseOdooFieldObservers()` is using the new `__odooFieldObserversToPause` [defined right before calling it](https://github.com/odoo/odoo/pull/176198/files#diff-aa4686cd9f6a0d50e1040ba3dda0b29e9898e7cc31a47f1f02b46a2c4c047a72R1058-R1064), which is something the code in [A] doesn't. So I think it would be better to keep "all observers" as a fallback for this "very specific" situation?